common/djangoapps/student/models.py

```diff
@@ -1437,6 +1460,8 @@ def enroll(cls, user, course_key, mode=None, check_access=False):
         if mode is None:
             mode = _default_course_mode(text_type(course_key))
         # All the server-side checks for whether a user is allowed to enroll.
+        # TODO: check if the given user has requested retirement
```

Comment:
How come we're not checking this now?

Reply:
How could the user be attempting to enroll after they've requested retirement? Is this check proposed just-in-case there's a loophole?

Reply:
This looks like something I forgot to implement, or decided against implementing.  I'll take another look.

Reply:
I'm going to leave this as is because it's a little too far off track from the original motivation for this PR (to protect new account registrations with retired email addresses).  I'll just make sure there's something in our testing spreadsheet to cover this case and we can cross that bridge when we get there.